recipes/openblas/all/conanfile.py

```diff
@@ -75,19 +83,28 @@ def layout(self):
     def generate(self):
         tc = CMakeToolchain(self)
 
-        if self.options.build_lapack:
-            self.output.warning("Building with lapack support requires a Fortran compiler.")
         tc.cache_variables["NOFORTRAN"] = not self.options.build_lapack
+        # This checks explicit user-specified fortran compiler
+        if self.options.build_lapack:
+            if not self._fortran_compiler:
+                if Version(self.version) < "0.3.21":
```

Comment:
This should probably be less than this version, as this is the version in which the fortran requirement is being removed
```suggestion
                if Version(self.version) < "0.3.24":
```